include/dblayer/mysql.php

```diff
@@ -182,7 +182,7 @@ function close()
 	{
 		if ($this->link_id)
 		{
-			if ($this->query_result)
+			if (is_object($this->query_result))
```

Comment:
I think [is_resource()](http://php.net/is_resource) would be a better fit here, since that is the actual return type of a valid response.